examples/README.md

```diff
@@ -65,6 +65,28 @@ Learn how to use hooks for data transformation and validation:
 - Conditional logic in hooks
 - Error handling and business rules
 
+### [07. JSON Schema Integration](./07-json-schema-integration.ts)
+
+Automatically generate factories from JSON Schema definitions:
+
+- Creating factories from JSON Schema objects
+- Supporting all JSON Schema types and formats
+- Enhanced schema composition with `allOf`, `anyOf`, `oneOf`
+- Custom format generators for domain-specific data
+- OpenAPI/Swagger schema compatibility
+- Advanced `$ref` handling with recursion prevention
+- Data validation against original schemas
+- Performance benchmarking vs manual factories
+- Integration with existing Factory features
+- Batch generation with `buildMany()` for API compatibility
+- Real-world API testing scenarios
+
+**Note**: This example requires optional dependencies:
+
+```bash
+npm install ajv ajv-formats
```

Comment:
These should be defined as optional peer dependencies in our package.json with a minimal version range